### ISSUE TITLE
Before steps into (separate) log groups

### ISSUE THREAD

**Issue by iomedico-beyer** [badeball/cypress-cucumber-preprocessor#922]
### Current behavior

[v15.1.0](https://github.com/badeball/cypress-cucumber-preprocessor/releases/tag/v15.1.0) puts test steps (`Given`, `When`, `Then`, `And`, `But`) into log groups, which is great! But all the `Before` stuff is still without log groups.

Also, when multiple `Before` statements do match, they cannot be distinguished in the log. The all just start by logging “Before”.

### Desired behavior

1. I would love to have one log group for each matching `Before` statement for a given test scenario. 
2. The group name should also reflect if the `Before` matches by tag name.

### Test code to reproduce

```
Before(function () {
    cy.log('TODO: login');
});

Before({ tags: '@deleteFoo' }, function () {
    cy.log('TODO: delete foo');
});
```

### Versions

* **Cypress version**: 12.3.0
* **Preprocessor version**: 15.1.0
* **Node version**: 16.16.0

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
Hey @iomedico-beyer, I've fixed this with v15.1.1.

**Comment by iomedico-beyer** [comment]
Amazing! 

The only thing is that the second point „The group name should also reflect if the `Before` matches by tag name.“ doesn’t seem to work. The group name is always just „Before“. For the code above with `yarn cypress open` it looks like this:

![image](https://user-images.githubusercontent.com/78728837/212344736-f43f9ee5-8cf0-4273-9bf0-7434e026bfbf.png)

**Comment by badeball** [comment]
Good idea! Since the requirements are getting a little bit more complex, I think it's warranted that it's tested as well. I think I've seen Cypress themselves testing somewhat similar, where they in a Cypress test actually verifies the runner's appearance. I'd like to dig through their test to find that and replicate it, although I can't right now. If you or anyone finds time though, feel free to do that.

**Comment by badeball** [comment]
This has been fixed in v17.0.0.
